modules/tinymce/src/themes/silver/test/ts/browser/editor/bespoke/DropdownAriaLabelTest.ts

```diff
@@ -0,0 +1,113 @@
+import { UiFinder } from '@ephox/agar';
+import { afterEach, describe, it } from '@ephox/bedrock-client';
+import { Attribute, SugarBody } from '@ephox/sugar';
+import { TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+import * as MenuUtils from '../../../module/MenuUtils';
+
+interface Scenario {
+  readonly menuLabel: string;
+  readonly initialItem: string;
+  readonly finalItem: string;
+}
+
+describe('browser.tinymce.themes.silver.editor.bespoke.DropdownAriaLabelTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    content_css: '/project/tinymce/src/themes/silver/test/css/content.css',
+    toolbar: 'align fontfamily fontsize blocks styles'
+  });
+
+  const testDropdownAriaLabel = (scenario: Scenario) => async () => {
```

Comment:
There are a whole lot of `if` checks and branches here that will make it harder to understand which path the code has followed if a test fails.

I think it's been refactored too heavily. Instead of a single function, please make a series of functions (it looks like you'll need 3?) so there are no branches in the code. This will initially be heavily duplicated, which is fine, you can then look at where it makes sense to extract pieces into helper functions.

Some duplication is fine - I've grown fond of saying "duplication is documentation". This doesn't mean avoid refactoring, but strategic duplication will greatly aid readability which is essential to tracking down problems when things fail.

It's particularly important for tests like this where you can't have good error messages. If a menu name is wrong all you get is `waited for element blah to appear, and it didn't`. This doesn't tell you what the incorrect value was - it can't - so the code needs to be easy to follow otherwise failures can require debugging the test.

Reply:
That's a good point, I've kind of had a feeling that having so many branches impacted readability. I've refactored it into 3 functions that pass callback functions to a common testing function which has no branches.